MemoryMeter.c

```diff
@@ -35,26 +37,28 @@ static void MemoryMeter_updateValues(Meter* this) {
 
    Settings *settings = this->host->settings;
 
-   /* shared, compressed and available memory are not supported on all platforms */
-   this->values[MEMORY_METER_SHARED] = NAN;
-   this->values[MEMORY_METER_COMPRESSED] = NAN;
-   this->values[MEMORY_METER_AVAILABLE] = NAN;
+   /* not all memory classes are supported on all platforms */
+   for (unsigned int memoryClassIdx = 0; memoryClassIdx < Platform_numberOfMemoryClasses; memoryClassIdx++) {
+      this->values[memoryClassIdx] = NAN;
+   }
+
    Platform_setMemoryValues(this);
+   this->curItems = (uint8_t) Platform_numberOfMemoryClasses;
+
+   /* compute the used memory */
+   double used = 0.0;
+   for (unsigned int memoryClassIdx = 0; memoryClassIdx < Platform_numberOfMemoryClasses; memoryClassIdx++) {
+      if (Platform_memoryClasses[memoryClassIdx].countsAsUsed)
+         used += this->values[memoryClassIdx];
+   }
```

Comment:
Prefer with all braces to be consistent with below:
```suggestion
   for (unsigned int memoryClassIdx = 0; memoryClassIdx < Platform_numberOfMemoryClasses; memoryClassIdx++) {
      if (Platform_memoryClasses[memoryClassIdx].countsAsUsed) {
         used += this->values[memoryClassIdx];
      }
   }
```